pulsar-broker/src/main/java/org/apache/pulsar/broker/service/ServerCnx.java

```diff
@@ -382,19 +382,24 @@ private boolean invalidOriginalPrincipal(String originalPrincipal) {
     // ////
 
     private CompletableFuture<Boolean> isTopicOperationAllowed(TopicName topicName, TopicOperation operation,
-                    AuthenticationDataSource authData) {
+                    AuthenticationDataSource authDataSource, AuthenticationDataSource originalAuthDataSource) {
         if (!service.isAuthorizationEnabled()) {
             return CompletableFuture.completedFuture(true);
         }
         CompletableFuture<Boolean> isProxyAuthorizedFuture;
-        if (originalPrincipal != null) {
-            isProxyAuthorizedFuture = service.getAuthorizationService().allowTopicOperationAsync(
-                topicName, operation, originalPrincipal, authData);
+        if (service.getPulsar().getConfiguration().isAuthenticateOriginalAuthData()) {
```

Comment:
There looks to have breaking change.

When `authenticateOriginalAuthData() == false`, we shouldn't check the original permission, which is client role, not proxy role.

Reply:
Revert to old code.